libbeat/scripts/cmd/global_fields/main_test.go

```diff
@@ -0,0 +1,792 @@
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "metricbeat", "module", "zookeeper", "mntr", "_meta", "fields.yml"),
+			Indent: 8,
+		},
+	}
+	packetbeatFiles = []*fields.YmlFile{
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "amqp", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "cassandra", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "dns", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "http", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "icmp", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "memcache", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "mongodb", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "mysql", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "nfs", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "pgsql", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "redis", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "thrift", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+		&fields.YmlFile{
+			Path:   filepath.Join(beatsPath, "packetbeat", "protos", "tls", "_meta", "fields.yml"),
+			Indent: 0,
+		},
+	}
+)
+
+// TestCollectModuleFiles validates if the required files are collected
+func TestCollectModuleFiles(t *testing.T) {
+	cases := []testcase{
+		testcase{
+			fieldsPath: filepath.Join(beatsPath, "filebeat", "module"),
+			files:      filebeatFiles,
+		},
+		testcase{
+			fieldsPath: filepath.Join(beatsPath, "heartbeat", "monitors", "active"),
+			files:      heartbeatFiles,
+		},
+		testcase{
+			fieldsPath: filepath.Join(beatsPath, "libbeat", "processors"),
+			files:      libbeatFiles,
+		},
+		testcase{
+			fieldsPath: filepath.Join(beatsPath, "metricbeat", "module"),
+			files:      metricbeatFiles,
+		},
+		testcase{
+			fieldsPath: filepath.Join(beatsPath, "packetbeat", "protos"),
+			files:      packetbeatFiles,
+		},
+	}
+
+	for _, c := range cases {
+		fieldFiles, err := fields.CollectModuleFiles(c.fieldsPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.True(t, reflect.DeepEqual(fieldFiles, c.files))
```

Comment:
Is this going to fail every time we add or remove a `fields.yml` file? Could we have a custom testcase that doesn't depend on the actual files?

Reply:
No, it doesn't. It does not run on CI checks. You need to run it manually if you change anything to the fields.yml collector. I am not a huge fan of writing files and creating directories for a unit test, so I settled for this.